blockly/tasks/messages.js

```diff
@@ -6,80 +6,74 @@ module.exports = function(grunt) {
-      }
+      var backend_init = loadBackend(language);
+      var backend = backend_init.backend;
+      var mf = backend_init.mf;
 
       // Generate javascript message functions.
-      files.forEach(function(file) {
-        var src = file.src[0];
-        var translated = grunt.file.readJSON(src);
-        //XXX Shouldn't have to re-read source strings for each locale, nor
-        // should this code be doing weird path replace munging.
-        var source = grunt.file.readJSON(
-          src.replace('build/', '').replace(locale, 'en_us'));
-        var code = 'var MessageFormat = require("messageformat");';
-        code += backend;
-        Object.keys(source).forEach(function(key) {
-          code += 'exports.' + key + ' = ';
-          try {
-            if (translated[key]) {
-              code += mf.precompile(mf.parse(translated[key]));
-            } else {
-              code += mf.precompile(mf.parse(source[key]));
-            }
-          } catch (e) {
-            // Fallback to English on failure.
-            grunt.log.warn('Failed to compile ' + src + ':' + key + '\n' + e);
+      var translated = grunt.file.readJSON(src);
+      var source = grunt.file.readJSON(src);
+      var code = 'var MessageFormat = window.messageformat;';
+      code += backend;
+      Object.keys(source).forEach(function(key) {
+        code += 'exports.' + key + ' = ';
+        try {
+          if (translated[key]) {
+            code += mf.precompile(mf.parse(translated[key]));
+          } else {
             code += mf.precompile(mf.parse(source[key]));
           }
-          code += ';\n\n';
-        });
-        grunt.file.write(file.dest, code);
+        } catch (e) {
+          // Fallback to English on failure.
+          grunt.log.warn('Failed to compile ' + src + ':' + key + '\n' + e);
+          code += mf.precompile(mf.parse(source[key]));
+        }
+        code += ';\n\n';
       });
-    });
 
+      // browserify module pre+postamble
+      var namespace = src.indexOf('common') > -1 ? 'locale' : 'appLocale';
+      var prepend = '!function(e){if("object"==typeof exports&&"undefined"!=typeof module)module.exports=e();else if("function"==typeof define&&define.amd)define([],e);else{var o;"undefined"!=typeof window?o=window:"undefined"!=typeof global?o=global:"undefined"!=typeof self&&(o=self),(o.blockly||(o.blockly={})).'+namespace+'=e()}}(function(){var define,module,exports;return (function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error("Cannot find module "+o+"");throw f.code="MODULE_NOT_FOUND",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({1:[function(require,module,exports){';
+      var append = '},{"messageformat":"messageformat"}]},{},[1])(1)});';
```

Comment:
@wjordan Why hardcode the pre+postamble?  Won't these change when browserify updates?